PythonVisualizations/Heap.py

```diff
@@ -0,0 +1,284 @@
+        self.startAnimations()       
+        
+        #If array needs to grow, add cells:
+        if self.size < len(self.list) + 1:
+            #making a tag for the array 
+            arrayList = self.canvas.find_withtag("arrayCell")
+            #making a tag for the copy of the array 
+            for cell in arrayList:
+                item = self.copyCanvasItem(cell)
+                self.canvas.itemconfig(item, tags="newArrayCell")
+            newArrayList = self.canvas.find_withtag("newArrayCell")
+            newArrayList = list(newArrayList)
+            boxes = list(newArrayList)
+            callEnviron |= set(boxes)   
+            #adding the values to the list of the copy of the array 
+            for i in range(len(self.list)):
+                newArrayList += [self.list[i].display_shape, self.list[i].display_val]
+            self.moveItemsBy(newArrayList, (2*self.CELL_WIDTH, 0))          #moving the whole array over
+            
+            # Growing the the array 
+            for i in range(self.size*2): 
+                self.createArrayCell(i)            
+            self.size *= 2
+            #copying the values back into the larger array 
+            copyVals = []
+            for i in range(len(self.list)):
+                copyVals += [self.list[i].display_shape, self.list[i].display_val]
+            self.moveItemsBy(copyVals, (-2*self.CELL_WIDTH, 0))    
+            #getting rid of the smaller array 
+            
+
+        #don't move arrow up if the first cell is being filled because it is already pointing there
+        if len(self.list) >= 1:
+            self.moveItemsBy(self.indexDisplay, (0, + (self.CELL_HEIGHT)))
+
+        cellCoords = self.cellCoords2(len(self.list)) # Color box
+        cellCenter = self.cellCenter2(len(self.list)) # Number in box
+
+        # create new cell and cell value display objects
+        cellCoords = add_vector(cellCoords, (0, 0, 0, self.CELL_BORDER))
+        toPositions = (cellCoords, cellCenter)
+
+        # determine the top left and bottom right positions
+        startPosition = [self.HEAP_X0, 0, self.HEAP_X0, 0] #Color 
+        startPosition = add_vector(startPosition, (0, 0, self.CELL_WIDTH, self.CELL_HEIGHT)) #color`
+        cellPair = self.createCellValue(startPosition, val)
+        callEnviron.add(cellPair)
+        self.moveItemsTo(cellPair, toPositions, steps=self.CELL_HEIGHT, sleepTime=0.01)
+    
+        # add a new Drawable with the new value, color, and display objects
+        d = drawable(
+            val, self.canvas.itemconfigure(cellPair[0], 'fill')[-1], *cellPair)
+        self.list.append(d)    #store item at the end of the list     
+        
+        if len(self.list) <= 0:            # heap condition.  The root node, i = 0,
+            return         
+        
+        for i in range(len(self.list)-1, 0, -1):
+            if self.list[i] < self.list[(i - 1) // 2]:  #if i is less than its parent 
```

Comment:
The heap, according to issue #75 , is supposed to be a max heap, but it is functioning as a min heap. We think just switching the `<` to a `>` should do the trick